sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs

```diff
@@ -361,6 +361,10 @@ private static string GetCanonicalName(string account, string containerName, str
         /// </summary>
         private void EnsureState()
         {
+            if (ExpiresOn == default || string.IsNullOrEmpty(Permissions))
+            {
+                throw Errors.SasMissingData();
```

Comment:
Please throw separate errors.  It's a lot less frustrating for users.  Use `nameof(ExpiresOn)`, etc., as well so refactoring keeps things updated.